src/hotspot/share/classfile/klassFactory.cpp

```diff
@@ -173,15 +173,16 @@ InstanceKlass* KlassFactory::create_from_stream(ClassFileStream* stream,
   assert(loader_data != NULL, "invariant");
   assert(THREAD->is_Java_thread(), "must be a JavaThread");
 
-  ResourceMark rm(THREAD);
-  HandleMark hm(THREAD);
+  JavaThread* current = THREAD->as_Java_thread();
```

Comment:
You can delete the assert at line 174 as the same assert is inside as_Java_thread().

But I think this is the kind of change that Coleen objects to: introducing a new local variable just to avoid using THREAD. The changes in this file can probably be reverted.

Reply:
Ok. I've reverted the changes.